repose-aggregator/components/filters/header-translation-filter/src/main/scala/org/openrepose/filters/headertranslation/HeaderTranslationFilter.scala

```diff
@@ -74,10 +74,19 @@ class HeaderTranslationFilter @Inject()(configurationService: ConfigurationServi
         if (originalHeaderValues.nonEmpty) {
           sourceHeader.getNewName foreach { newHeaderName =>
             originalHeaderValues foreach { originalHeaderValue =>
-              if (quality.isEmpty) {
-                httpRequest.addHeader(newHeaderName, originalHeaderValue)
+              def writeHeaders(writeWithoutQuality: (String, String) => Unit,
+                               writeWithQuality: (String, String, Double) => Unit): Unit = {
+                if (quality.isEmpty) {
+                  writeWithoutQuality(newHeaderName, originalHeaderValue)
+                } else {
+                  writeWithQuality(newHeaderName, withoutQuality(originalHeaderValue), quality.get)
+                }
+              }
```

Comment:
So to say it clearer, the two things aren't really related and this allows me to separate them and deal with them individually.